package/kedro_viz/integrations/kedro/data_loader.py

```diff
@@ -114,3 +120,33 @@ def load_data(
 
     context = load_context(project_path=project_path, env=env)
     return context.catalog, context.pipelines
+
+
+def load_data_for_all_versions(
+    filepath: str, limit: int = 10
+) -> Optional[Dict[datetime, Any]]:
+    """Load data for all versions of the dataset
+    Args:
+        filepath: the path where the dataset is located.
+        limit: the maximum number of past versions we want to load.
+    Returns:
+        A dictionary containing the version and the json data inside each version
+    """
+    version_list = [
+        path for path in sorted(Path(filepath).iterdir(), reverse=True) if path.is_dir()
+    ]
+    versions = {}
+    for version in version_list[:limit]:
+        try:
+            timestamp = datetime.strptime(version.name, VERSION_FORMAT)
+        except ValueError:
+            logger.warning(
+                """Expected timestamp of format YYYY-MM-DDTHH:MM:SS.ffffff.
+                 Skip when loading metrics."""
+            )
+            continue
+        else:
+            path = version / Path(filepath).name
+            with open(path) as fs_file:
+                versions[timestamp] = json.load(fs_file)
```

Comment:
We will need to catch an exception here as well if the data can't be loaded with `json`.